portfolio/src/main/webapp/apiLibrary.js

```diff
@@ -0,0 +1,177 @@
+const CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+var GoogleAuth;
+
+// Constants for discovery documents' URLs.
+const DISCOVERY_DOCS_CALENDAR =
+    'https://www.googleapis.com/discovery/v1/apis/calendar/v3/rest';
+
+const DISCOVERY_DOCS_SHEETS =
+    'https://sheets.googleapis.com/$discovery/rest?version=v4';
+
+const DISCOVERY_DOCS_TASKS =
+    'https://www.googleapis.com/discovery/v1/apis/tasks/v1/rest';
+
+// Scopes for API access
+const SCOPE_CALENDAR_READ_ONLY =
+    'https://www.googleapis.com/auth/calendar.readonly';
+const SCOPE_CALENDAR_READ_WRITE = 'https://www.googleapis.com/auth/calendar';
+const SCOPE_SHEETS_READ_WRITE = 'https://www.googleapis.com/auth/spreadsheets';
+const SCOPE_TASKS_READ_ONLY = 'https://www.googleapis.com/auth/tasks.readonly';
+
+// Object for all error codes
+const ERROR_CODES = {
+  popup_closed_by_user: 'popup_closed_by_user',
+  access_denied: 'access_denied'
+};
+
+function fetchApiKey() {
+  return fetch('./appConfigServlet').then(response => response.json());
+}
+
+/**
+ * Loads the API's client and auth2 modules.
+ * Calls the initCalendarClient function after the modules load.
+ */
+function handleClientLoad() {
+  gapi.load('client:auth2', initAuthentication);
+}
+
+function initAuthentication() {
+  fetchApiKey().then(
+      responseJson => gapi.client
```

Comment:
Consider using a helper method for gapi.client.init call (it can take responseJson as a param)

Reply:
Done.